sdks/python/src/opik/evaluation/metrics/heuristics/bleu.py

```diff
@@ -0,0 +1,218 @@
+                reason="Candidate is empty"
+            )
+
+        # 2) Process references
+        if isinstance(reference, str):
+            if not reference.strip():
+                return score_result.ScoreResult(
+                    value=0.0,
+                    name=self.name,
+                    reason="Reference is empty"
+                )
+            references = [reference.lower().split()]
+        else:
+            references = []
+            for ref in reference:
+                if not ref.strip():
+                    return score_result.ScoreResult(
+                        value=0.0,
+                        name=self.name,
+                        reason="Reference is empty"
+                    )
+                references.append(ref.lower().split())
+
+        candidate = output.lower().split()
+
+        # Truncate & normalize weights to the candidate length
+        used_weights = self._truncate_weights(len(candidate))
+
+        smoothing_func = self._get_smoothing_func()
+
+        try:
+            bleu_value = sentence_bleu(
+                references,
+                candidate,
+                weights=used_weights,
+                smoothing_function=smoothing_func
+            )
+        except ZeroDivisionError:
+            # edge case if references or candidate is basically empty after splitting
+            bleu_value = 0.0
+
+        return score_result.ScoreResult(
+            value=bleu_value,
+            name=self.name,
+            reason=f"Sentence-level BLEU (nltk, method={self.smoothing_method}): {bleu_value:.4f}",
+        )
+
+    ###########################################################################
+    # CORPUS-LEVEL BLEU
+    ###########################################################################
+    def score_corpus(
+        self,
+        outputs: List[str],
+        references_list: List[Union[str, List[str]]],
+        **ignored_kwargs: Any
+    ) -> score_result.ScoreResult:
+        """
+        Computes a corpus-level BLEU score using nltk.translate.bleu_score.corpus_bleu.
+        """
+
+        if len(outputs) != len(references_list):
+            return score_result.ScoreResult(
+                value=0.0,
+                name=self.name,
+                reason="Mismatch: number of candidates != number of references.",
+            )
+
+        all_candidates = []
+        all_references = []
+
+        for output, ref_item in zip(outputs, references_list):
+            if not output.strip():
+                # If candidate is empty, skip it (leading to zero or ignoring).
+                continue
+
+            candidate_tokens = output.lower().split()
+
+            if isinstance(ref_item, str):
+                if not ref_item.strip():
+                    continue
+                refs = [ref_item.lower().split()]
+            else:
+                refs = []
+                skip_this = False
+                for r in ref_item:
```

Comment:
I'd suggest improve naming and don't use such shortcuts, it affects readability.
`ref_item` doesn't tell that it's a collection of some values, it already looks like an item in collection.
Then what is `r` in `ref_item`? ref item item? :)